### PR TITLE
Bug 1636842 - part 2: Make check-screenshots script smarter about the number of screenshots

### PR DESCRIPTION
**Pull request by JohanLorenzo**

### CODE REVIEW

File: l10n-screenshots-config.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    - tr
+    - uk
+    - ur
+    - uz
+    - vi
+    - zgh
+    - zh-CN
+    - zh-TW
+locales-with-shorter-context-menu: ["hi-IN", "th", "zh-CN", "zh-TW"]  # These locales have 1 less screenshot. 
+usual-number-of-screenshots: 55
```
Comment (JohanLorenzo) [inline l10n-screenshots-config.yml:95]: ⚠️ If we add/remove tests, then we must update this value.

File: l10n-screenshots-config.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    - tl
+    - tr
+    - uk
+    - ur
+    - uz
+    - vi
+    - zgh
+    - zh-CN
+    - zh-TW
+locales-with-shorter-context-menu: ["hi-IN", "th", "zh-CN", "zh-TW"]  # These locales have 1 less screenshot.
```
Comment (JohanLorenzo) [inline l10n-screenshots-config.yml:94]: I think capturing edge cases by defining a list of locales that behave the same way is more maintainable. I thought of making a file that tells how many screenshots each single locale has, but that'll be cumbersome and prone to error to update.

File: l10n-screenshots.sh [hunk 3]
```diff
@@ -25,13 +25,14 @@ fi
...
 if [ "$1" = '--test-without-building' ]; then
-  EXTRA_FAST_LANE_ARGS='--test_without_building'
-  shift
+    EXTRA_FAST_LANE_ARGS='--test_without_building'
+    shift
 fi
 
 LOCALES=$*
 if [ $# -eq 0 ]; then
-  LOCALES=$(cat "$PROJECT_DIR/l10n-screenshots-locales.txt")
```
Comment (JohanLorenzo) [inline l10n-screenshots.sh:34]: It's not easy to parse a Yaml file in bash and I don't think we want to run all locales by default anymore. So I updated the script to force users to tell which locales they want to run. How does this sound to you @isabelrios?

File: taskcluster/scripts/check-screenshots.py [hunk 4]
```diff
@@ -36,37 +38,56 @@ def _init_logging():
     )
 
 
+def _parse_screenshots_configuration(config_file):
+    config = yaml.safe_load(config_file)
+
+    number_of_screenshots_per_locale = {}
+    for locale in config["locales"]:
```
Comment (JohanLorenzo) [inline taskcluster/scripts/check-screenshots.py:45]: This loop is in charge of counting how many screenshots a locale has. As of now a locale can have either 55 or 54 pictures. We may have other kinds of locales that bump/reduce this number. Each new kind will have to be defined here.

File: taskcluster/scripts/check-screenshots.py [hunk 5]
```diff
@@ -36,37 +38,56 @@ def _init_logging():
...
     if archives != expected_archives:
         errors.append(
             "The list of archives (zip files) does not match the expected one. Expected: {}. Got: {}.".format(
                 expected_archives, archives
             )
         )
 
-    log.info("Processing {} archives...".format(len(archives)))
+    log.info("Processing {} archives...".format(len(expected_archives)))
+    log.debug("Archives are expected to have these numbers of screenshots: {}".format(expected_number_of_screenshots_per_archive))
```
Comment (JohanLorenzo) [inline taskcluster/scripts/check-screenshots.py:79]: This way we know what the script expects.

File: l10n-screenshots.sh [hunk 6]
```diff
@@ -13,8 +13,8 @@ PROJECT_DIR="$(get_abs_path $CURRENT_DIR/../../../..)"
 
 
 if [ -d l10n-screenshots ]; then
-  echo "The l10n-screenshots directory already exists. You decide."
-  exit 1
+    echo "The l10n-screenshots directory already exists. You decide."
+    exit 1
```
Comment (isabelrios) [inline l10n-screenshots.sh:17]: nit: extra space? also above and some lines below

File: l10n-screenshots.sh (hunk 3 above)
Comment (isabelrios) [reply l10n-screenshots.sh:34]: sounds good, that would only affect when running them locally, right? on CI, this is passed each time if  I got this right

File: l10n-screenshots.sh (hunk 6 above)
Comment (JohanLorenzo) [reply l10n-screenshots.sh:17]: yeah, I aligned every if statement to use 4-space-indents. The rest of the file seems to use this convention. I'm fine having everything aligned at 2 spaces too 🙂

File: l10n-screenshots.sh (hunk 3 above)
Comment (JohanLorenzo) [reply l10n-screenshots.sh:34]: That's right!

**Pull request merged**